### PR TITLE
Migrate project to angular-cli

### PR DESCRIPTION
**Pull request by marjan-georgiev**
**What kind of change does this PR introduce?** (check one with "x")
- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [x] CI related changes
- [ ] Other... Please describe:

**What is the new behavior?**
Migrates the project to angular-cli
Builds using ng-packagr
Ivy compatible
Removes the compiled files from the git repository
Reformats the whole codebase

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: projects/swimlane/ngx-datatable/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "@swimlane/ngx-datatable",
+  "version": "15.0.2",
+  "description": "ngx-datatable is an Angular table grid component for presenting large and complex data.",
+  "peerDependencies": {
```
Comment (luisb90) [inline projects/swimlane/ngx-datatable/package.json]: I believe you have to add the dependencies as well to the library's package.json, right?

File: projects/swimlane/ngx-datatable/src/lib/components/body/body.component.ts [hunk 2]
```diff
@@ -298,50 +283,60 @@ export class DataTableBodyComponent implements OnInit, OnDestroy {
...
 
   /**
    * Called once, before the instance is destroyed.
    */
   ngOnDestroy(): void {
-    if (this.rowDetail) this.listener.unsubscribe();
-    if (this.groupHeader) this.listener.unsubscribe();
+    if (this.rowDetail) {
+      this.listener.unsubscribe();
+    }
```
Comment (luisb90) [inline projects/swimlane/ngx-datatable/src/lib/components/body/body.component.ts]: can combine these into one if statement with an `||`

File: projects/swimlane/ngx-datatable/src/lib/components/datatable.component.ts [hunk 3]
```diff
@@ -541,9 +471,7 @@ export class DatatableComponent implements OnInit, DoCheck, AfterViewInit {
    */
   @HostBinding('class.fixed-row')
   get isFixedRow(): boolean {
-    const rowHeight: number | string = this.rowHeight;
-    return (typeof rowHeight === 'string') ?
-      (<string>rowHeight) !== 'auto' : true;
+    return this.rowHeight !== 'auto';
```
Comment (luisb90) [inline projects/swimlane/ngx-datatable/src/lib/components/datatable.component.ts]: a thought, if the function they pass to rowHeight is impure, the row could potentially not be fixed and this check wont catch it

File: projects/swimlane/ngx-datatable/src/lib/components/header/header.component.ts [hunk 4]
```diff
@@ -58,25 +67,25 @@ import { MouseEvent } from '../../events';
...
   @Input() set innerWidth(val: number) {
     this._innerWidth = val;
-
-    if (this._columns) {    
-      const colByPin = columnsByPin(this._columns);
-      this._columnGroupWidths = columnGroupWidths(colByPin, this._columns);
-      this.setStylesByGroup();
-    }
+    setTimeout(() => {
+      if (this._columns) {
```
Comment (luisb90) [inline projects/swimlane/ngx-datatable/src/lib/components/header/header.component.ts]: can you skip the setTimeout by moving the if check outside of it? or is the timeout to allow `this._columns` to populate?

File: projects/swimlane/ngx-datatable/package.json (hunk 1 above)
Comment (marjan-georgiev) [reply projects/swimlane/ngx-datatable/package.json]: Not unless the library has any dependencies, afaik.

File: projects/swimlane/ngx-datatable/src/lib/components/header/header.component.ts (hunk 4 above)
Comment (marjan-georgiev) [reply projects/swimlane/ngx-datatable/src/lib/components/header/header.component.ts]: Yup, it basically skips the cycle allowing _columns to populate. This is a change in behavior caused by Ivy.

File: projects/swimlane/ngx-datatable/src/lib/components/datatable.component.ts (hunk 3 above)
Comment (marjan-georgiev) [reply projects/swimlane/ngx-datatable/src/lib/components/datatable.component.ts]: The value would still be provided from outside though. As far as the datatable is concerned, it doesn't have to calculate the row height itself.

Comment (luisb90) [reply projects/swimlane/ngx-datatable/src/lib/components/datatable.component.ts]: makese sense, gotcha

**Comment by luisb90** [comment]
found a disparity issue while running through the demos.
http://swimlane.github.io/ngx-datatable/#server-scrolling
in the above, when you scroll down, it shows a blue loading bar across the bottom of the table, as more records are loaded. Locally, while the records do load, the loading bar does not show.

**Comment by marjan-georgiev** [comment]
Fixed - needed to update the style in that demo.

**Comment by bmesin** [comment]
Are you sure it works now?

image

**Comment by hugoj-goncalves** [comment]
> Are you sure it works now?
> 
> image

even on wannabebrowser (I.E. 9) seems fine to me...

**Comment by marjan-georgiev** [comment]
> Are you sure it works now?
> 
> image

What are you testing on?

This is what I see:

image

**Comment by bmesin** [comment]
I usually use Chrome and it's ok there, but this was on Safari.

**Pull request merged**